kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/CsvLineProcessorFactory.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2019 The Hyve
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.radarbase.connect.upload.converter
+
+import org.radarbase.connect.upload.api.ContentsDTO
+import org.radarbase.connect.upload.api.RecordDTO
+
+/**
+ * Processor for processing single lines of CSV file.
+ */
+interface CsvLineProcessorFactory {
+    /** Upper case header list. */
+    val header: List<String>
+
+    /**
+     * Whether the file contents matches this CSV line processor.
+     */
+    fun matches(contents: ContentsDTO) = contents.fileName.endsWith(".csv")
+
+    /**
+     * Whether the header matches this CSV line processor. The provided header must be in upper
+     * case.
+     */
+    fun matches(header: List<String>) = header.containsAll(this.header)
+
+    /**
+     * Create a line processor for given record.
+     */
+    fun createLineProcessor(record: RecordDTO, logRepository: LogRepository): CsvLineProcessor
+
+    interface CsvLineProcessor {
+        val recordLogger: RecordLogger
+
+        /**
+         * Whether the given line is valid. If not, the line will be discarded. An error can be
+         * thrown to mark the entire file invalid.
+         */
+        fun isLineValid(
+                header: List<String>,
+                line: Array<String>): Boolean {
+            return when {
+                line.isEmpty() -> {
+                    recordLogger.warn("Empty line found ${line.toList()}")
+                    false
+                }
+                header.size != line.size -> {
+                    recordLogger.warn("Line size ${line.size} does not match with header size ${header.size}. Skipping this line")
+                    false
+                }
+                line.any { it.isEmpty() } -> {
```

Comment:
Is this correct? I thought we also wanted to be able to parse lines with null (empty) values on it.

Reply:
Yes. Good point. I think we should remove this check and allow null. However, we should also add some null check before parsing the objects to data types (if null anticipated), which we don't do yet I think.

Reply:
@blootsvoets Could we take this on the next release? I am a little reluctant since we haven't tested this. I expect another release soon when we have finished the  on axivity implementation. We already have an issue for it https://github.com/RADAR-base/radar-upload-source-connector/issues/106